lib/validates_zipcode/cldr_regex_collection.rb

```diff
@@ -132,117 +134,120 @@ module CldrRegexpCollection
       PF: /\A987\d{2}\z/,
       PG: /\A\d{3}\z/,
       PM: /\A9[78]5\d{2}\z/,
-      PN: /\APCRN 1ZZ\z/,
+      PN: /\APCRN[ ]?1ZZ\z/i,
       PW: /\A96940\z/,
-      RE: /\A9[78]4\d{2}\z/,
-      SH: /\A(ASCN|STHL) 1ZZ\z/,
+      RE: /\A((97)(4|7|8)[0-9][0-9])\z/,
+      SH: /\A(ASCN|STHL|TDCU)[ ]?1ZZ\z/i,
       SJ: /\A\d{4}\z/,
-      SZ: /\A[HLMS]\d{3}\z/,
-      TC: /\ATKCA 1ZZ\z/,
+      SZ: /\A[HLMS]\d{3}\z/i,
+      TC: /\ATKCA[ ]?1ZZ\z/i,
```

Comment:
SZ: case insensitive, just need a verdict on this one.
TC: Another GB postcode format